tests/e2e/validatecluster/validatecluster_test.go

```diff
@@ -1,73 +1,257 @@
+				}
+				ip, _ := e2e.FetchNodeExternalIP(nodeName)
+				cmd := "curl  --header host:foo1.bar.com" + " http://" + ip + "/name.html"
+				fmt.Println(cmd)
+
+				Eventually(func(g Gomega) {
+					res, _ := e2e.RunCommand(cmd)
+					fmt.Println(res)
+					g.Expect(res).Should(ContainSubstring("test-ingress"))
+				}, "240s", "5s").Should(Succeed())
+			}
+		})
+
+		It("Verify Daemonset", func() {
+			_, err := e2e.DeployWorkload("daemonset.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println(err, "daemonset manifest not deployed")
+			}
+			nodes, _ := e2e.ParseNodes(kubeConfigFile, false)
+			pods, _ := e2e.ParsePods(kubeConfigFile, false)
+
+			Eventually(func(g Gomega) {
+				count := e2e.CountOfStringInSlice("test-daemonset", pods)
+				fmt.Println("POD COUNT")
+				fmt.Println(count)
+				fmt.Println("NODE COUNT")
+				fmt.Println(len(nodes))
+				g.Expect(len(nodes)).Should((Equal(count)), "Daemonset pod count does not match node count")
+			}, "240s", "10s").Should(Succeed())
 		})
+
+		It("Verify dns access", func() {
+			_, err := e2e.DeployWorkload("dnsutils.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println(err, "dnsutils manifest not deployed")
+			}
+
+			Eventually(func(g Gomega) {
+				cmd := "kubectl --kubeconfig=" + kubeConfigFile + " exec -i -t dnsutils -- nslookup kubernetes.default"
+				fmt.Println(cmd)
+				res, _ := e2e.RunCommand(cmd)
+				fmt.Println(res)
+				g.Expect(res).Should(ContainSubstring("kubernetes.default.svc.cluster.local"))
+			}, "240s", "2s").Should(Succeed())
+		})
+
+		It("Verify Local Path Provisioner storage ", func() {
+			_, err := e2e.DeployWorkload("local-path-provisioner.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println(err, "local-path-provisioner manifest not deployed")
+			}
+
+			Eventually(func(g Gomega) {
+				cmd := "kubectl get pvc local-path-pvc --kubeconfig=" + kubeConfigFile
+				res, _ := e2e.RunCommand(cmd)
+				fmt.Println(res)
+				g.Expect(res).Should(ContainSubstring("local-path-pvc"))
+				g.Expect(res).Should(ContainSubstring("Bound"))
+			}, "240s", "2s").Should(Succeed())
+
+			Eventually(func(g Gomega) {
+				cmd := "kubectl get pod volume-test --kubeconfig=" + kubeConfigFile
+				res, _ := e2e.RunCommand(cmd)
+				fmt.Println(res)
+
+				g.Expect(res).Should(ContainSubstring("volume-test"))
+				g.Expect(res).Should(ContainSubstring("Running"))
+			}, "420s", "2s").Should(Succeed())
+
+			cmd := "kubectl --kubeconfig=" + kubeConfigFile + " exec volume-test -- sh -c 'echo local-path-test > /data/test'"
+			_, _ = e2e.RunCommand(cmd)
+			fmt.Println("Data stored in pvc: local-path-test")
+
+			cmd = "kubectl delete pod volume-test --kubeconfig=" + kubeConfigFile
+			res, _ := e2e.RunCommand(cmd)
+			fmt.Println(res)
+
+			_, err = e2e.DeployWorkload("local-path-provisioner.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println(err, "local-path-provisioner manifest not deployed")
+			}
+
+			time.Sleep(1 * time.Minute)
```

Comment:
I don't think we should have large sleep times in a test. If you need more wait time, extend the Eventually timeout.
```suggestion

```